pkg/util/util.go

```diff
@@ -0,0 +1,27 @@
+package util
+
+import (
+	"strings"
+
+	"github.com/openshift/csi-driver-shared-resource/pkg/config"
+)
+
+// IsAuthorizedSharedResource verify the shared resource already exists under OCP pre-populated shared resource
+func IsAuthorizedSharedResource(kind, resource, resourceRefKeyNm, resourceRefKeyNs string) bool {
+	var (
+		value  string
+		exists bool
+	)
+	if strings.HasPrefix(resource, "openshift-") {
+		switch kind {
+		case "SharedSecret":
+			value, exists = config.LoadedSharedSecrets.AuthorizedSharedResources[resource]
+		case "SharedConfigMap":
+			value, exists = config.LoadedSharedConfigMaps.AuthorizedSharedResources[resource]
+		}
+		if !exists || (exists && value != (resourceRefKeyNs+":"+resourceRefKeyNm)) {
```

Comment:
```suggestion
		if !exists || value != (resourceRefKeyNs+":"+resourceRefKeyNm) {
```
If the first part of the expression is false `!exists` it won't even check the second part, so there is no need for the `exists` check.